packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -954,6 +954,28 @@ abstract class CustomClipper<T> {
   String toString() => '$runtimeType';
 }
 
+/// A [CustomClipper] that clips to the outer path of a [ShapeBorder].
+class ShapeBorderClipper extends CustomClipper<Path> {
+  /// Creates a [ShapeBorder] clipper.
+  ///
+  /// The [shapeBorder] argument must not be null.
+  const ShapeBorderClipper({
+    @required this.shapeBorder
+  }) : assert(shapeBorder != null);
+
+  // The shape border whose outer path this clipper clips to.
+  final ShapeBorder shapeBorder;
+
+  /// Returns the outer path of [shapeBorder] as the clip.
+  @override
+  Path getClip(Size size) => shapeBorder.getOuterPath(Offset.zero & size);
```

Comment:
this needs to pass in the text direction

Reply:
done